---
layout: default
title: Primers and barcodes for 16S amplification
---
---
**Primers and barcodes for 16S amplification**

!TOC


####16S and ITS amplification.

Using the extracted DNA as a template, primer sequences[^1],[^2] containing 10bp unique [barcodes] and 454 FLX Titanium adaptors were used to perform PCR amplification of the internal transcribed spacer (ITS) region of the fungal rDNA and the 16S V3-V5 region of bacteria rDNA.  
<br/>

##### Barcodes
||||||
|--------|-|-|-|
|ACATAGTAGT|ACGCACACGT|ACGTATGACT|ACGTCTCGCT|
|ACTATATCTA|ACTCTAGTCT|ACTCTGTGAG|ACTGTATATG|
|ACTGTCTGTA|AGACATATCG|AGACGTCTCT|AGAGCGCGAG|
|AGAGTACGTG|AGAGTCACGA|AGATACACAG|AGCAGCTACG|
|AGCGTCAGTA|AGTATCTCAG|AGTCAGTGTA|AGTCGACATG|
|AGTCTAGTAG|AGTCTCTAGA|AGTGACTAGT|ATACGCACGT|
|ATACTGCGTA|ATATACGCGT|ATCGCTAGTA|ATCGTAGTGT|
|CACGATAGCG|CAGTGAGACG|CATACTACTA|CATATAGACG|
|CGACTACGAG|CGCGCACGCG|CGCGCTACGT|CGTACTCAGT|
|CGTAGATCGA|CGTCAGCACG|CGTGAGTGCG|CGTGTCGCAG|
|CTACAGAGAG|CTACATCACG|CTACGTCTGT|CTAGAGTGTG|
|CTAGCTCGTA|CTAGTATGCG|CTAGTGCACG|CTCACGCGAG|
|CTCGTGACGT|CTCTGTCTCG|CTGTACGTAG|TACACGATGT|
|TACGAGTACA|TAGCACACTA|TAGCGTGATG|TAGCTACTGT|
|TAGTGTAGCG|TATACACGCG|TATAGCGTCT|TATATCGACA|
|TATCACTACG|TATCTGTATA|TATGATACTA|TCACACTATA|
|TCACGACGTA|TCAGCGTCTA|TCAGTCTCGA|TCATCGAGTG|
|TCGACGAGCT|TCGACGCATA|TCGAGAGTCG|TCGAGCACGT|
|TCGCTACGCG|TCGCTATATA|TCGCTGATAG|TCTACAGCGT|
|TCTATACGCT|TCTATCTAGT|TCTCGTGTGT|TCTCTCGAGA|
|TCTGTGACAG|TGACGTGACA|TGAGACGAGA|TGAGATCTCG|
|TGAGTATGAG|TGATGACGCG|TGCGTACTCA|TGCTACTGAG|
|TGCTGCGTCG|TGTACGCACG|TGTACTACAT|TGTAGAGTAG|
|TGTCGTAGCA|TGTCTGCTAG|||
||||

<br/>
#### 16S Bacterial Primers || V3-V5 Region

| Direction | Primer| Sequence |
|----------|-------------|-----------|
| Forward | 341F | 5’-CCTACGGGAGGCAGCAG-3’ |
| Reverse | 926R | 5’-CCGTCAATTCMTTTRAGT-3’|

<br/>
#### ITS Fungal Primers || Nested Strategy

For fungal ITS amplification, we employed a nested strategy involving an initial outer amplification with the primer set, NSA3 and NLC2, followed by the amplification of the resulting sequences by the primer set ITS1F and ITS4A.[^3],[^4],[^5]

| Direction | Primer| Sequence |
|----------|-------------|-----------|
|Nested Forward  | NSA3 | 5'-AAACTCTGTCGTGCTGGGGATA-3'|  
|Nested Reverse | NLC2 | 5'-GAGCTGCATTCCCAAACAACTC-3' |
|Forward  | ITS1F | 5’-CTTGGTCATTTAGAGGAAGTAA-3’      |
|Reverse | ITS4A | 5’-CGCCGTTACTGGGGCAATCCCTG-3’      |

Following amplification, the PCR products were cleaned using the Qiagen MinElute 96 UF PCR Purification Kit and quantified using the Invitrogen Quant-DNA Assay Kit, High Sensitivity. 25ng of amplified DNA was pooled from each sample, concentrated and cleaned up to remove any residual primers or nucleotides using the Agencourt AMPure XP Kit. The amplicons were sequenced on the 454 FLX Titanium sequencing platform.
<br/>

#### References

[^1]: Muyzer G, de Waal EC, Uitterlinden AG. 1993. Profiling of complex microbial populations by denaturing gradient gel electrophoresis analysis of polymerase chain reaction-amplified genes coding for 16S rRNA. Appl. Environ. Microbiol. 59:695–700.

[^2]: Lane DJ. 1991. 16S/23S rRNA sequencing. Nucleic acid techniques in bacterial systematics 125–175.

[^3]: Martin KJ, Rygiewicz PT. 2005. Fungal-specific PCR primers developed for analysis of the ITS region of environmental DNA extracts. BMC Microbiol. 5:28.

[^4]: Larena I, Salazar O, González V, Julián MC, Rubio V. 1999. Design of a primer for ribosomal DNA internal transcribed spacer with enhanced specificity for Ascomycetes. J. Biotechnol. 75:187–194.

[^5]: Gardes M, Bruns TD. 1993. ITS primers with enhanced specificity for Basidiomycetes--application to the identification of mycorrhizae and rusts. Mol. Ecol. 2:113–118.

<!--Google Analytics Code-->
[![Analytics](https://ga-beacon.appspot.com/UA-59204692-1/bayesianmice/gh-pages/pcr_primers?pixel)](https://github.com/igrigorik/ga-beacon)
