<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en-us">
<head>
    <!-- meta attributes -->
    <title>The YAP Workflow</title>
    <meta name="author" content="Open Pencil" />
    <meta name="description" content="Bayesian modeling for C. albicans colonization.">

    <!-- MathJax -->
    <script type="text/javascript"
            src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">

    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
        }
        });

    MathJax.Hub.Queue(function() {
      var all = MathJax.Hub.getAllJax(), i;
      for(i=0; i < all.length; i += 1) {
        all[i].SourceElement().parentNode.className += ' has-jax';
        }
        });
    </script>

   <!-- syntax highlighting CSS -->
   <link rel="stylesheet" href="/css/syntax.css" type="text/css" />

   <!-- Homepage CSS -->
   <link rel="stylesheet" href="/css/screen.css" type="text/css" media="screen, projection" />
</head>
<body>

<!-- ClickTale Top part -->
<script type="text/javascript">
var WRInitTime=(new Date()).getTime();
</script>
<!-- ClickTale end of Top part -->

<div class="site">
  <div class="title">
    <a href="..//" class="site-avatar"><img src="https://raw.githubusercontent.com/openpencil/openpencil.github.io/master/assets/OpenPencil.png" /></a>
      <a href="/">bayesianmice</a>
      
      <a class="extra" href="/data.html">Data</a>
      
      
      <a class="extra" href="/methods.html">Methods</a>
      
      
      <a class="extra" href="/results.html">Results</a>
      
      <a class="extra" href="https://github.com/openpencil/bayesianmice">Code</a>


  </div>
  <hr />

<p><strong>The YAP Bioinformatics Workflow Engine</strong></p>

<p><ul>
 <li>
 <ul>
  <li>
  <ul>
   <li>
   <ul>
    <li>
    <ul>
     <li><a href="#Computationally.efficient.16S.and.ITS.Reads.Processing.and.Taxonomic.Classification">Computationally efficient 16S and ITS Reads Processing and Taxonomic Classification</a>
     <ul>
      <li><a href="#Preliminary.Processing.and.Quality.Checks">Preliminary Processing and Quality Checks</a></li>
      <li><a href="#Clustering.and.Taxonomic.Assignment">Clustering and Taxonomic Assignment</a></li>
     </ul>
     </li>
     <li><a href="#Resolution.of.unclassified.sequences">Resolution of unclassified sequences</a></li>
     <li><a href="#L16S.and.ITS.Databases.">16S and ITS Databases.</a></li>
    </ul>
    </li>
    <li><a href="#References">References</a></li>
   </ul>
   </li>
  </ul>
  </li>
 </ul>
 </li>
</ul>

</p>

<a name="Computationally.efficient.16S.and.ITS.Reads.Processing.and.Taxonomic.Classification"></a>
<h5>Computationally efficient 16S and ITS Reads Processing and Taxonomic Classification</h5>

<p>Taxonomic classification of the 16S and ITS sequences was performed using <a href="https://github.com/shpakoo/YAP" title="Szpakowski S. YAP: A Computationally Efficient Workflow for Taxonomic Analyses of Bacterial 16S and Fungal ITS Sequences . GitHub; 2013.">YAP</a>, a computationally optimized bioinformatics workflow<sup id="fnref:1"><a href="#fn:1" rel="footnote">1</a></sup>.  The workflow was run on a distributed computing cluster managed by the <a href="http://www.univa.com/products/grid-engine" title="Univa Products: Grid Engine Software for Workload Scheduling and Management.">Univa Grid Engine</a>.</p>

<a name="Preliminary.Processing.and.Quality.Checks"></a>
<h6>Preliminary Processing and Quality Checks</h6>

<p>Within YAP, the 454 sequencer-generated SFF files were converted into FASTA and qual files using <em>sffinfo</em> from the Roche SFF Tools software package. After demultiplexing the sequences with mothur<sup id="fnref:2"><a href="#fn:2" rel="footnote">2</a></sup>, quality filtering of the sequences was performed using mothur’s implementation of PyroNoise<sup id="fnref:3"><a href="#fn:3" rel="footnote">3</a></sup>,<sup id="fnref:4"><a href="#fn:4" rel="footnote">4</a></sup>. Sequences shorter than 220 bases were discarded<sup id="fnref:5"><a href="#fn:5" rel="footnote">5</a></sup> and sequences spanning the V4 and the V5 regions of the 16S rDNA were retained. Duplicate sequences were collapsed and 454 hybrid sequences were removed with CD-HIT-454<sup id="fnref:6"><a href="#fn:6" rel="footnote">6</a></sup>,<sup id="fnref:7"><a href="#fn:7" rel="footnote">7</a></sup>. The sequences were then aligned against the 16S SILVA database to verify their 16S origins and to ensure that the sequences were correctly positioned on the V4 and V5 regions<sup id="fnref:8"><a href="#fn:8" rel="footnote">8</a></sup>. Using the uchime implementation in mothur and the 16S SILVA database, multi-organism hybrid-sequences and chimeras were discarded<sup id="fnref:9"><a href="#fn:9" rel="footnote">9</a></sup>. Segments with coverage lower than 90% of the maximum coverage were trimmed from the sequences.</p>

<a name="Clustering.and.Taxonomic.Assignment"></a>
<h6>Clustering and Taxonomic Assignment</h6>

<p>CD-HIT-EST, a module of the CD-HIT suite, was used to cluster the sequences at the 97% similarity level to produce operational taxonomic units (OTU)<sup id="fnref:10"><a href="#fn:10" rel="footnote">10</a></sup>,<sup id="fnref:11"><a href="#fn:11" rel="footnote">11</a></sup>. Taxonomical classification of the final set of reads was performed using mothur’s implementation of a naive Bayes classifier<sup id="fnref:12"><a href="#fn:12" rel="footnote">12</a></sup> trained on a normalized RDP training dataset<sup id="fnref:13"><a href="#fn:13" rel="footnote">13</a></sup>. Sequence counts for any given taxon were obtained by summing the cluster sizes of all OTUs that were classified as that taxon.</p>

<a name="Resolution.of.unclassified.sequences"></a>
<h5>Resolution of unclassified sequences</h5>

<p>For OTUs that remained unclassified at the genus level after processing through YAP, genus labels were assigned using the NCBI Megablast algorithm<sup id="fnref:14"><a href="#fn:14" rel="footnote">14</a></sup>. Final genus labels were the ones with the lowest E-value. In addition, for sequences assigned to a given genus of interest, for example, Candida, the most predominant species-level taxonomic labels were identified using the NCBI Megablast algorithm<sup id="fnref:15"><a href="#fn:15" rel="footnote">15</a></sup>. Predominant species were the ones that were most frequent within a particular genus, belonged to the largest OTU clusters, occurred in the largest number of mice samples and obtained the lowest E-value in the Megablast algorithm.</p>

<a name="L16S.and.ITS.Databases."></a>
<h5>16S and ITS Databases.</h5>

<p>In the YAP workflow, high quality databases were used to a) verify the origin of the 16S and ITS sequences, b) check for chimeras and c) assign taxonomies. For bacterial 16S processing and assignment, the SILVA database<sup id="fnref:16"><a href="#fn:16" rel="footnote">16</a></sup> and a training dataset from the RDP project<sup id="fnref:17"><a href="#fn:17" rel="footnote">17</a></sup> were used. For ITS, a custom database was built as described below. A seed database of the entire fungal ribosomal region was first constructed. This seed database consisted of 236 diverse ribosomal sequences and was obtained by running NT BLAST on a 7785 nucleotide long, annotated ribosomal region of Aspergillus fumigatus Af293<sup id="fnref:18"><a href="#fn:18" rel="footnote">18</a></sup>,<sup id="fnref:19"><a href="#fn:19" rel="footnote">19</a></sup>. The seed database was aligned using T-coffee<sup id="fnref:20"><a href="#fn:20" rel="footnote">20</a></sup>. Next, a total of 155,136 unique ITS sequences were retrieved from the NCBI public database. The program MUSCLE was used to align these ITS sequences to the seed database and orient them in the 5’ to 3’ direction <sup id="fnref:21"><a href="#fn:21" rel="footnote">21</a></sup>. Over 5% of sequences were re-oriented. Based on the alignment scores, either the original or the reverse complements of these sequences were retained for further processing. To improve taxonomic classification, identical sequences as well as substrings were clustered using CD-HIT in two iterations <sup id="fnref:22"><a href="#fn:22" rel="footnote">22</a></sup>. The first iteration, with 100% sequence identity, yielded 149,266 clusters. The second iteration, using 97% sequence identity, yielded 39,548 species level clusters. After two iterations of clustering, annotations were reassigned for 2,153 sequences. A total of 10,226 sequences were still taxonomically identified as “uncultured”. In the final step, the NCBI sequence IDs of the properly oriented exemplars of the clusters were queried against the NCBI taxonomy database to obtain the maximum possible resolution of taxonomic levels for the poorly annotated sequences <sup id="fnref:23"><a href="#fn:23" rel="footnote">23</a></sup>.</p>

<!--Websites-->


<a name="References"></a>
<h4>References</h4>
<div class="footnotes">
<hr/>
<ol>
<li id="fn:1">
<p>Szpakowski S. YAP: A Computationally Efficient Workflow for Taxonomic Analyses of Bacterial 16S and Fungal ITS Sequences. GitHub. 2013.<a href="#fnref:1" rev="footnote">&#8617;</a></p></li>
<li id="fn:2">
<p>Schloss PD, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, et al. Introducing mothur: open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol. American Society for Microbiology; 2009 Dec 1;75(23):7537–41. Available from: http://dx.doi.org/10.1128/AEM.01541-09<a href="#fnref:2" rev="footnote">&#8617;</a></p></li>
<li id="fn:3">
<p>Quince C, Lanzen A, Davenport RJ, Turnbaugh PJ. Removing noise from pyrosequenced amplicons. BMC Bioinformatics . 2011 Jan 28;12(1):38+. Available from: http://dx.doi.org/10.1186/1471-2105-12-38<a href="#fnref:3" rev="footnote">&#8617;</a></p></li>
<li id="fn:4">
<p>Schloss PD, Gevers D, Westcott SL. Reducing the effects of PCR amplification and sequencing artifacts on 16S rRNA-based studies. PLoS One . Public Library of Science; 2011 Dec 14;6(12):e27310+. Available from: http://dx.doi.org/10.1371/journal.pone.0027310<a href="#fnref:4" rev="footnote">&#8617;</a></p></li>
<li id="fn:5">
<p>Engelbrektson A, Kunin V, Wrighton KC, Zvenigorodsky N, Chen F, Ochman H, et al. Experimental factors affecting PCR-based estimates of microbial species richness and evenness. ISME J . Nature Publishing Group; 2010 May 21;4(5):642–7. Available from: http://dx.doi.org/10.1038/ismej.2009.153<a href="#fnref:5" rev="footnote">&#8617;</a></p></li>
<li id="fn:6">
<p>Huang Y, Niu B, Gao Y, Fu L, Li W. CD-HIT Suite: a web server for clustering and comparing biological sequences. Bioinformatics . Oxford University Press; 2010 Mar 1;26(5):680–2. Available from: http://dx.doi.org/10.1093/bioinformatics/btq003<a href="#fnref:6" rev="footnote">&#8617;</a></p></li>
<li id="fn:7">
<p>Niu B, Fu L, Sun S, Li W. Artificial and natural duplicates in pyrosequencing reads of metagenomic data. BMC Bioinformatics . BioMed Central; 2010 Apr 13;11(1):1–11. Available from: http://dx.doi.org/10.1186/1471-2105-11-187<a href="#fnref:7" rev="footnote">&#8617;</a></p></li>
<li id="fn:8">
<p>Pruesse E, Quast C, Knittel K, Fuchs BM, Ludwig W, Peplies J, et al. SILVA: a comprehensive online resource for quality checked and aligned ribosomal RNA sequence data compatible with ARB. Nucleic Acids Res . Oxford University Press; 2007 Oct 18;35(21):7188–96. Available from: http://dx.doi.org/10.1093/nar/gkm864<a href="#fnref:8" rev="footnote">&#8617;</a></p></li>
<li id="fn:9">
<p>Haas BJ, Gevers D, Earl AM, Feldgarden M, Ward DV, Giannoukos G, et al. Chimeric 16S rRNA sequence formation and detection in Sanger and 454-pyrosequenced PCR amplicons. Genome Res . Cold Spring Harbor Laboratory Press; 2011 Mar 1;21(3):494–504. Available from: http://dx.doi.org/10.1101/gr.112730.110<a href="#fnref:9" rev="footnote">&#8617;</a></p></li>
<li id="fn:10">
<p>Huang Y, Niu B, Gao Y, Fu L, Li W. CD-HIT Suite: a web server for clustering and comparing biological sequences. Bioinformatics . Oxford University Press; 2010 Mar 1;26(5):680–2. Available from: http://dx.doi.org/10.1093/bioinformatics/btq003<a href="#fnref:10" rev="footnote">&#8617;</a></p></li>
<li id="fn:11">
<p>Hamady M, Knight R. Microbial community profiling for human microbiome projects: Tools, techniques, and challenges. Genome Res . Cold Spring Harbor Laboratory Press; 2009 Jul 1;19(7):1141–52. Available from: http://dx.doi.org/10.1101/gr.085464.108<a href="#fnref:11" rev="footnote">&#8617;</a></p></li>
<li id="fn:12">
<p>Wang Q, Garrity GM, Tiedje JM, Cole JR. Na"ive Bayesian Classifier for Rapid Assignment of rRNA Sequences into the New Bacterial Taxonomy. Appl Environ Microbiol . Center for Microbial Ecology, Michigan State University, East Lansing, MI 48824, USA.: American Society for Microbiology; 2007 Aug 15;73(16):5261–7. Available from: http://dx.doi.org/10.1128/aem.00062-07<a href="#fnref:12" rev="footnote">&#8617;</a></p></li>
<li id="fn:13">
<p>Lan Y, Wang Q, Cole JR, Rosen GL. Using the RDP classifier to predict taxonomic novelty and reduce the search space for finding novel organisms. PLoS One . 2012 Mar 5;7(3):e32491. Available from: http://dx.doi.org/10.1371/journal.pone.0032491<a href="#fnref:13" rev="footnote">&#8617;</a></p></li>
<li id="fn:14">
<p>Camacho C, Coulouris G, Avagyan V, Ma N, Papadopoulos J, Bealer K, et al. BLAST+: architecture and applications. BMC Bioinformatics . 2009 Dec 15;10(1):421+. Available from: http://dx.doi.org/10.1186/1471-2105-10-421<a href="#fnref:14" rev="footnote">&#8617;</a></p></li>
<li id="fn:15">
<p>Camacho C, Coulouris G, Avagyan V, Ma N, Papadopoulos J, Bealer K, et al. BLAST+: architecture and applications. BMC Bioinformatics . 2009 Dec 15;10(1):421+. Available from: http://dx.doi.org/10.1186/1471-2105-10-421<a href="#fnref:15" rev="footnote">&#8617;</a></p></li>
<li id="fn:16">
<p>Quast C, Pruesse E, Yilmaz P, Gerken J, Schweer T, Yarza P, et al. The SILVA ribosomal RNA gene database project: improved data processing and web-based tools. Nucleic Acids Res . Oxford University Press; 2013 Jan 1;41(Database issue):D590–D596. Available from: http://dx.doi.org/10.1093/nar/gks1219<a href="#fnref:16" rev="footnote">&#8617;</a></p></li>
<li id="fn:17">
<p>Cole JR, Wang Q, Cardenas E, Fish J, Chai B, Farris RJ, et al. The Ribosomal Database Project: improved alignments and new tools for rRNA analysis. Nucleic Acids Res . Oxford University Press; 2009 Jan 1;37(Database issue):D141–D145. Available from: http://dx.doi.org/10.1093/nar/gkn879<a href="#fnref:17" rev="footnote">&#8617;</a></p></li>
<li id="fn:18">
<p>Arnaud MB, Cerqueira GC, Inglis DO, Skrzypek MS, Binkley J, Chibucos MC, et al. The Aspergillus Genome Database (AspGD): recent developments in comprehensive multispecies curation, comparative genomics and community resources. Nucleic Acids Res . Oxford University Press; 2012 Jan 1;40(Database issue):D653–D659. Available from: http://dx.doi.org/10.1093/nar/gkr875<a href="#fnref:18" rev="footnote">&#8617;</a></p></li>
<li id="fn:19">
<p>Camacho C, Coulouris G, Avagyan V, Ma N, Papadopoulos J, Bealer K, et al. BLAST+: architecture and applications. BMC Bioinformatics . 2009 Dec 15;10(1):421+. Available from: http://dx.doi.org/10.1186/1471-2105-10-421<a href="#fnref:19" rev="footnote">&#8617;</a></p></li>
<li id="fn:20">
<p>Notredame C, Higgins DG, Heringa J. T-Coffee: A novel method for fast and accurate multiple sequence alignment. J Mol Biol . National Institute for Medical Research, The Ridgeway, London, NW7 1AA, UK. cedric.notredame@europe.com; 2000 Sep 8;302(1):205–17. Available from: http://dx.doi.org/10.1006/jmbi.2000.4042<a href="#fnref:20" rev="footnote">&#8617;</a></p></li>
<li id="fn:21">
<p>Edgar RC. MUSCLE: multiple sequence alignment with high accuracy and high throughput. Nucleic Acids Res . bob@drive5.com: Oxford University Press; 2004 Mar 19;32(5):1792–7. Available from: http://dx.doi.org/10.1093/nar/gkh340<a href="#fnref:21" rev="footnote">&#8617;</a></p></li>
<li id="fn:22">
<p>Huang Y, Niu B, Gao Y, Fu L, Li W. CD-HIT Suite: a web server for clustering and comparing biological sequences. Bioinformatics . Oxford University Press; 2010 Mar 1;26(5):680–2. Available from: http://dx.doi.org/10.1093/bioinformatics/btq003<a href="#fnref:22" rev="footnote">&#8617;</a></p></li>
<li id="fn:23">
<p>Benson DA, Karsch-Mizrachi I, Lipman DJ, Ostell J, Sayers EW. GenBank. Nucleic Acids Res . 2009 Jan 1;37Database issue):D26–31. Available from: http://dx.doi.org/10.1093/nar/gkn723<a href="#fnref:23" rev="footnote">&#8617;</a></p></li>
</ol>
</div>


  <ul class="posts_list">
      
    </ul>

  <div class="footer">
    <div class="contact">
      <p>
        <a href="http://github.com/openpencil/bayesianmice"><svg width="20px" height="20px" viewBox="0 0 60 60" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:sketch="http://www.bohemiancoding.com/sketch/ns">
    <path d="M0.336871032,30 C0.336871032,13.4314567 13.5672313,0 29.8877097,0 C46.208188,0 59.4385483,13.4314567 59.4385483,30 C59.4385483,46.5685433 46.208188,60 29.8877097,60 C13.5672313,60 0.336871032,46.5685433 0.336871032,30 Z M0.336871032,30" id="Github" fill="#333333" sketch:type="MSShapeGroup"></path>
    <path d="M18.2184245,31.9355566 C19.6068506,34.4507902 22.2845295,36.0156764 26.8007287,36.4485173 C26.1561023,36.9365335 25.3817877,37.8630984 25.2749857,38.9342607 C24.4644348,39.4574749 22.8347506,39.62966 21.5674303,39.2310659 C19.7918469,38.6717023 19.1119377,35.1642642 16.4533306,35.6636959 C15.8773626,35.772144 15.9917933,36.1507609 16.489567,36.4722998 C17.3001179,36.9955141 18.0629894,37.6500075 18.6513541,39.04366 C19.1033554,40.113871 20.0531304,42.0259813 23.0569369,42.0259813 C24.2489236,42.0259813 25.0842679,41.8832865 25.0842679,41.8832865 C25.0842679,41.8832865 25.107154,44.6144649 25.107154,45.6761142 C25.107154,46.9004355 23.4507693,47.2457569 23.4507693,47.8346108 C23.4507693,48.067679 23.9990832,48.0895588 24.4396415,48.0895588 C25.3102685,48.0895588 27.1220883,47.3646693 27.1220883,46.0918317 C27.1220883,45.0806012 27.1382993,41.6806599 27.1382993,41.0860982 C27.1382993,39.785673 27.8372803,39.3737607 27.8372803,39.3737607 C27.8372803,39.3737607 27.924057,46.3153869 27.6704022,47.2457569 C27.3728823,48.3397504 26.8360115,48.1846887 26.8360115,48.6727049 C26.8360115,49.3985458 29.0168704,48.8505978 29.7396911,47.2571725 C30.2984945,46.0166791 30.0543756,39.2072834 30.0543756,39.2072834 L30.650369,39.1949165 C30.650369,39.1949165 30.6837446,42.3123222 30.6637192,43.7373675 C30.6427402,45.2128317 30.5426134,47.0792797 31.4208692,47.9592309 C31.9977907,48.5376205 33.868733,49.5526562 33.868733,48.62514 C33.868733,48.0857536 32.8436245,47.6424485 32.8436245,46.1831564 L32.8436245,39.4688905 C33.6618042,39.4688905 33.5387911,41.6768547 33.5387911,41.6768547 L33.5988673,45.7788544 C33.5988673,45.7788544 33.4186389,47.2733446 35.2190156,47.8992991 C35.8541061,48.1209517 37.2139245,48.1808835 37.277815,47.8089257 C37.3417055,47.4360167 35.6405021,46.8814096 35.6252446,45.7236791 C35.6157088,45.0178155 35.6567131,44.6059032 35.6567131,41.5379651 C35.6567131,38.470027 35.2438089,37.336079 33.8048426,36.4323453 C38.2457082,35.9766732 40.9939527,34.880682 42.3337458,31.9450695 C42.4383619,31.9484966 42.8791491,30.5737742 42.8219835,30.5742482 C43.1223642,29.4659853 43.2844744,28.1550957 43.3168964,26.6025764 C43.3092677,22.3930799 41.2895654,20.9042975 40.9014546,20.205093 C41.4736082,17.0182425 40.8060956,15.5675121 40.4961791,15.0699829 C39.3518719,14.6637784 36.5149435,16.1145088 34.9653608,17.1371548 C32.438349,16.3998984 27.0982486,16.4712458 25.0957109,17.3274146 C21.4005522,14.6875608 19.445694,15.0918628 19.445694,15.0918628 C19.445694,15.0918628 18.1821881,17.351197 19.1119377,20.6569598 C17.8961113,22.2028201 16.9902014,23.2968136 16.9902014,26.1963718 C16.9902014,27.8297516 17.1828264,29.2918976 17.6176632,30.5685404 C17.5643577,30.5684093 18.2008493,31.9359777 18.2184245,31.9355566 Z M18.2184245,31.9355566" id="Path" fill="#FFFFFF" sketch:type="MSShapeGroup"></path>
    <path d="M59.4385483,30 C59.4385483,46.5685433 46.208188,60 29.8877097,60 C23.8348308,60 18.2069954,58.1525134 13.5216148,54.9827754 L47.3818361,5.81941103 C54.6937341,11.2806503 59.4385483,20.0777973 59.4385483,30 Z M59.4385483,30" id="reflec" fill-opacity="0.08" fill="#000000" sketch:type="MSShapeGroup"></path>
</svg>
</a>
      </p>
    </div>
  </div>
</div>


</body>
</html>
